File: restore.py
import subprocess
try:
    with open('d:/Projects/RoadFireWall/frontend/src/components/Dashboard.tsx', 'wb') as f:
        f.write(subprocess.check_output(['git', 'show', 'HEAD:frontend/src/components/Dashboard.tsx'], cwd='d:/Projects/RoadFireWall'))
    print('Restore successful')
except Exception as e:
    print('Error:', e)

File: restore.js
const fs = require('fs');
const cp = require('child_process');
try {
  const out = cp.execSync('git show HEAD:frontend/src/components/Dashboard.tsx');
  fs.writeFileSync('D:/Projects/RoadFireWall/dashboard_old.txt', out);
} catch (e) {
  fs.writeFileSync('D:/Projects/RoadFireWall/dashboard_old.txt', String(e));
}
